api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -775,7 +775,13 @@ type FelixConfigurationSpec struct {
 
 	// WireguardEnabledV6 controls whether Wireguard is enabled for IPv6 (encapsulating IPv6 traffic over an IPv6 underlay network). [Default: false]
 	WireguardEnabledV6 *bool `json:"wireguardEnabledV6,omitempty"`
-	// WireguardThreadingEnabled controls whether Wireguard has NAPI threading enabled. [Default: false]
+	// WireguardThreadingEnabled controls whether Wireguard has Threaded NAPI enabled. [Default: false]
+	// This increases the maximum number of packets a Wireguard interface can process.
+	// There is a known issue https://lore.kernel.org/netdev/CALrw=nEoT2emQ0OAYCjM1d_6Xe_kNLSZ6dhjb5FxrLFYh4kozA@mail.gmail.com/T/ with this setting
+	// that may cause NAPI to get stuck holding the global `rtnl_mutex` when a peer is removed.
+	// Wireguard peers are removed during node reboots.
+	// Kernels which include this patch: https://lore.kernel.org/netdev/20240228121000.526645-2-bigeasy@linutronix.de/ are able to recover after a node drain.
+	// This feature should only be considered if you have high packets per second workloads that are causing dropping packets due to a saturated `softirq` CPU core.
```

Comment:
```suggestion
	// WireguardThreadingEnabled controls whether Wireguard has Threaded NAPI enabled. [Default: false]
	// This increases the maximum number of packets a Wireguard interface can process.
	// Consider threaded NAPI only if you have high packets per second workloads that are causing dropping packets due to a saturated `softirq` CPU core.
	// There is a known issue https://lore.kernel.org/netdev/CALrw=nEoT2emQ0OAYCjM1d_6Xe_kNLSZ6dhjb5FxrLFYh4kozA@mail.gmail.com/T/ with this setting
	// that may cause NAPI to get stuck holding the global `rtnl_mutex` when a peer is removed.
	// Workaround: Make sure your Linux kernel includes this patch: https://lore.kernel.org/netdev/20240228121000.526645-2-bigeasy@linutronix.de/.
```

Some edits for clarity and concision. A few other ideas:
* If there's a patch for this, do we expect this to be fixed in most distros? For one, it would be useful to specify major distro versions that are clear (for example: run Ubuntu 20.02 or later). Second, if it's likely to get distributed widely, this note may go stale very quickly.
* I removed a clause from your original, but I assumed you meant the same thing when you spoke of node drain and node reboot. If you intend for them to be different, let me know.

Reply:
I included the Github link to the patch and it shows the tags that include the commit. It is in the main kernel so it may go stale.
Yes I meant the same thing. Your refactor of the docs looks good!